test/simulator/moves/reflecttype.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const assert = require('./../../assert');
+const common = require('./../../common');
+
+let battle;
+
+describe('Reflect Type', function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it('should fail when used against a Pokemon whose type is "???"', function () {
+		battle = common.createBattle();
+		battle.join('p1', 'Guest 1', 1, [{species: 'Arcanine', ability: 'intimidate', moves: ['burnup']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: 'Latias', ability: 'levitate', item: 'laggingtail', moves: ['reflecttype']}]);
+		assert.constant(() => battle.p2.active[0].getTypes(), () => battle.commitDecisions());
+	});
+
+	it('should ignore the "???" type when used against a Pokemon whose type contains "???" and a non-added type', function () {
+		battle = common.createBattle();
+		const p1 = battle.join('p1', 'Guest 1', 1, [{species: 'Latias', ability: 'levitate', item: 'laggingtail', moves: ['reflecttype', 'trickortreat']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: 'Moltres', ability: 'intimidate', moves: ['burnup']}]);
+		battle.commitDecisions();
+		assert.strictEqual(battle.p1.active[0].getTypes().length, 1);
+		assert.strictEqual(battle.p1.active[0].getTypes()[0], 'Flying');
```

Comment:
This should also be changed to a single assert

    assert.strictEqual(battle.p1.active[0].getTypes().join('/'), 'Flying');